staker/block_validator.go

```diff
@@ -215,6 +222,19 @@ func NewBlockValidator(
 	}
 	streamer.SetBlockValidator(ret)
 	inbox.SetBlockValidator(ret)
+	if config().MemoryFreeLimit != "" {
+		limit, err := resourcemanager.ParseMemLimit(config().MemoryFreeLimit)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse MemoryFreeLimit string from config: %w", err)
+		}
+		limtchecker, err := resourcemanager.NewCgroupsMemoryLimitCheckerIfSupported(limit)
+		if err != nil {
+			log.Warn("failed to create MemoryFreeLimitChecker, Cgroups V1 or V2 is unsupported")
+		}
```

Comment:
We don't want to have anything too loud here if it's not going to work on all system, and is on by default.
One way to do it would be to have the default value be "default" - which would mean 1GB, but "default" would only log.Warn if it fails and other settings would propagate the error and fail loading if the setting failed.

Reply:
Agreed. Thanks for the idea!